src/data/datasets/classification/image_classification.py

```diff
@@ -0,0 +1,132 @@
+                interface of transforms in `albumentations` library.
+            augment: Optional augment to be applied on a sample.
+                This should have the interface of transforms in `albumentations` library.
+            input_dtype: Data type of of the torch tensors related to the image.
+            target_dtype: Data type of of the torch tensors related to the target.
+            target_column: Name of the column that contains target labels.
+            grayscale: If True, image will be read as grayscale otherwise as RGB.
+            expand_rate: A multiplier that shows how many times the dataset will be larger than its real size.
+                Useful for small datasets.
+            test_mode: If True, only image without labels will be returned.
+            multilabel: If True, target labels is being converted to multihot vector for multilabel task.
+            num_classes: Number of classes for multilabel task(length of multihot target vector).
+            lazy_init_multilabel: If True, the target variable is converted to multihot when __getitem__ is called.
+        """
+        super().__init__(data_folder, path_to_datalist, transform, input_dtype, grayscale, test_mode, augment)
+
+        self.__target_column = target_column
+        self.__target_dtype = target_dtype
+        self.__expand_rate = expand_rate
+        self.__multilabel = multilabel
+
+        self.update_transform_targets({'input': 'image'})
+
+        if multilabel:
+            self.__num_classes = num_classes
+            self.__lazy_init_multilabel = lazy_init_multilabel
+            if not self.lazy_init_multilabel and not self.test_mode:
+                process_func = partial(self.process_multilabel, num_classes=self.num_classes)
+                self.csv[self.target_column] = self.csv[self.target_column].apply(process_func)
+
+    @property
+    def target_column(self) -> str:
+        return self.__target_column
+
+    @property
+    def target_dtype(self) -> str:
+        return self.__target_dtype
+
+    @property
+    def expand_rate(self) -> int:
+        return self.__expand_rate
+
+    @property
+    def multilabel(self) -> bool:
+        return self.__multilabel
+
+    @property
+    def num_classes(self) -> int:
+        return self.__num_classes
+
+    @property
+    def lazy_init_multilabel(self) -> bool:
+        return self.__lazy_init_multilabel
+
+    def __getitem__(self, idx: int) -> dict:
+        idx = idx // self.expand_rate
+        record = self.csv.iloc[idx]
+        image = self._read_image(record)
+        sample = {'input': image, 'index': idx}
+        sample = self.apply_transform(self.augment, sample)
+        sample = self.apply_transform(self.transform, sample)
+        sample['input'] = sample['input'].type(torch.__dict__[self.input_dtype])
+
+        if self.test_mode:
+            return sample
+
+        sample['target'] = record[self.target_column]
```

Comment:
Add method `process_multiclass` where you will check the class index to fit the range